src/Sprites/ArtReader.cpp

```diff
@@ -0,0 +1,140 @@
+#include "ArtReader.h"
+#include "ArtFile.h"
+#include "PaletteHeader.h"
+#include "../Streams/FileReader.h"
+#include <stdexcept>
+#include <vector>
+
+namespace ArtReader {
+	// Anonymous namespace to hold private methods
+	//namespace {
+	void ReadPalette(Stream::SeekableReader& seekableReader, ArtFile& artFile);
+	void ReadImageMetadata(Stream::SeekableReader& seekableReader, ArtFile& artFile);
+	void ReadAnimations(Stream::SeekableReader& seekableReader, ArtFile& artFile);
+	Animation ReadAnimation(Stream::SeekableReader& seekableReader);
+	Frame ReadFrame(Stream::SeekableReader& seekableReader);
+	//}
```

Comment:
I think these methods should become private static members of ArtFile.

Reply:
I'd like to figure out why I'm getting compilation errors when wrapping them in the anonymous namespace.

Reply:
Hmm, I see. It's because you need to wrap both the forward declarations and the definitions in an anonymous namespace. The commented out lines suggest only the forward declarations were wrapped. As such, the definitions will be in a different namespace, and so be defining entirely new functions, leaving the anonymous namespace functions undefined.

Still, I disagree that anonymous namespaces are the correct solution here.

If all you want is to split up the implementation, you can already do that with a class. Define all methods in the header. Implement reading methods in one file, writing methods in another file, and data manipulations in yet another file.

Reply:
Okay, thanks I'll try this out to get the anonymous namespaces working. I had thought about separating the write/read into separate files but belonging to the same class/header. I'm not thrilled about all the associated function prototypes filling up the class' header file. This might be a good compromise though.